testutils-macros/src/lib.rs

```diff
@@ -540,7 +540,7 @@ pub fn bdk_blockchain_tests(attr: TokenStream, item: TokenStream) -> TokenStream
 
                     test_client.generate(1, Some(wallet_addr));
 
-                    wallet.sync(noop_progress(), None).unwrap();
+                    wallet.sync(noop_progress(), None).unwrap().address;
```

Comment:
oops I think you added `.address` to the wrong line, it should be on 536 above:
```rust
let wallet_addr = wallet.get_address(New).unwrap().address;
```

Reply:
Ah you're right.

When I fix that I end up with a lot of broken tests, and it seems like the problem is coming from within the `testutils` macro:

```
test blockchain::electrum::generated_tests_local_electrs::test_sync_before_and_after_receive ... FAILED
test blockchain::electrum::generated_tests_local_electrs::test_sync_bump_fee_add_input ... FAILED
test blockchain::electrum::generated_tests_local_electrs::test_sync_bump_fee_add_input_no_change ... FAILED
test blockchain::electrum::generated_tests_local_electrs::test_sync_bump_fee_remove_change ... FAILED
test blockchain::electrum::generated_tests_local_electrs::test_sync_long_change_chain ... FAILED
```

```
---- blockchain::electrum::generated_tests_local_electrs::test_sync_before_and_after_receive stdout ----
thread 'blockchain::electrum::generated_tests_local_electrs::test_sync_before_and_after_receive' panicked at 'called `Result::unwrap()` on an `Err` value: Io(Os { code: 2, kind: NotFound, message: "No such file or directory" })', /home/futurepaul/.cargo/registry/src/github.com-1ecc6299db9ec823/bdk-testutils-0.4.0/src/lib.rs:304:85
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

It's unclear to me where this file stuff is happening. (My hunch is that addresses are being written somewhere and this API breakage broke that, but I can't quite pin it down).

Would be happy to pair on this if you hit me up on Discord. Feel a little in over my head messing with this macro.

Reply:
If you are testing locally these tests are probably breaking because they require the `bitcoindevkit/electrs` docker container to be running and some env variables to be set (see https://github.com/bitcoindevkit/bitcoin-regtest-box#local-bdk-testing). But I'm also happy to walk you through it. Will PM you on Discord and we'll get it working.